test/Serilog.Tests/Context/LogContextTests.cs

```diff
@@ -383,4 +401,9 @@ public bool IsCallable()
         }
     }
 #endif
+
+    class ForceNewThreadSyncContext : SynchronizationContext
```

Comment:
Mostly a personal opinion, but maybe having a `IDisposable` class would make the test code a bit more readable (less aync lambdas , and less indentation generally) , a bit like https://github.com/serilog/serilog/blob/dev/test/Serilog.Tests/Support/TemporarySelfLog.cs and its usage in https://github.com/serilog/serilog/blob/96bfe61c7e35271450062846374c13cce75b83e5/test/Serilog.Tests/Core/ChildLoggerKnownLimitationsTests.cs#L17 

It would be nice to have test code that looks like : 

```csharp
using(NewThreadSynchronizationContext.Start()){
  // the code that needs to be tested with that SynchronizationContext
}
```

Reply:
couldn't use await inside disposable block, see explanation above